Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -16,12 +16,21 @@
 
 #import "FIRInstanceID.h"
 
+#ifdef SWIFT_PACKAGE
+#import "Firebase/Core/Private/FIRAppInternal.h"
+#import "Firebase/Core/Private/FIRComponent.h"
+#import "Firebase/Core/Private/FIRComponentContainer.h"
+#import "Firebase/Core/Private/FIRLibrary.h"
+#import "GULAppEnvironmentUtil.h"
```

Comment:
I wonder, would it make sense to define a macro like `FIREBASE_IMPORT(header_option1, header_fallback_option)`? If we have one, we can easier make sure we don't miss an import for one of package managers.

Reply:
You can't import/include in a macro.

This pattern we're putting everywhere is really gross. I think we should find one style of import and make it work for all our build systems. Why not just use the imports you're using in the `#ifdef SWIFT_PACKAGE` everywhere?

Reply:
I plan to do such a PR to consolidate the imports before merging back to master. I'm using the SPM PRs for experimentation.  The general idea right now is to use module imports for public headers and repo relative paths for private imports.   I believe that works for everything except GoogleUtilities public headers since the module names are different in SPM because SPM doesn't support subspecs.

Reply:
Also the import strategy is intertwined with some still unanswered questions about the swift package manager integration including its missing private header support and whether or not we want to abandon supporting individual library versioning support with the packager.